src/main/java/org/mockbukkit/mockbukkit/entity/PlayerMock.java

```diff
@@ -2360,28 +2360,49 @@ public void setHealth(double health)
 			return;
 		}
 
+		DamageSource dmg;
+		String msg;
+		if (getLastDamageCause() != null && getLastDamageCause().getDamageSource().getDirectEntity() instanceof Player killer)
+		{
+			dmg = DamageSource.builder(DamageType.PLAYER_ATTACK).build();
+			setKiller(killer);
+			msg = getName() + " was slain by " + killer.getName();
```

Comment:
Up to you, but there's already a translated messages for death events and such. I stole a large translations file from a Minecraft server (legally). The file is in resources/translations, where there's a shit ton of messages you can choose between, just search for `death.attack.*`

Reply:
Example of usecase:
```java
TranslatableComponent translatableComponent = Component.translatable("argument.entity.selector.unknown", Component.text("@h"));
		assertEquals("Unknown selector type '@h'", PlainTextComponentSerializer.plainText().serialize(translatableComponent));
```

Reply:
How would I do that? what is the proper way to retrieve entries from there?

Reply:
Just find the right translation key, I would say "death.attack.mob" or "death.attack.player" would do the trick. And then you fill the placeholder text with the names of the players or something like that

Reply:
I leave this up to you because I think it'll cleaner to combine LivingEntityMock::setHealth & PlayerMock::setHealth as they share sooooo much. At that time you could also add the translatable message in there. Want me to make a ticket for that? (and I want to head to bed ;))

Reply:
I will just merge this, I can make the ticket as well :). Good night!